src/app/compiler/compiler.js

```diff
@@ -219,10 +282,14 @@ function Compiler (handleImportCall) {
     worker.postMessage({cmd: 'loadVersion', data: url})
   }
 
-  function gatherImports (files, target, importHints, cb) {
```

Comment:
I remember now, `target` was used for this import loop to avoid cycles or something like that.

Reply:
oh ok, does that needs to be added to jsonio as well?

Reply:
It is not related to json io but to the import process browser-solidity uses.

Reply:
hmm ok I don't really see how this is used yet, I will take a look.
Just wondering cause `target` was part of the input of the `compile()`

Reply:
But the actual solc interface is `compileInternal` and it was not passed to that one.

Reply:
`target` is passed to `compileJSONCallback`